Network/DNS/Resolver.hs

```diff
@@ -173,13 +173,17 @@ makeResolvSeed conf = do
 
 getDefaultDnsServers :: FilePath -> IO [String]
 #if defined(WIN)
+foreign import ccall "getWindowsDefDnsServers" getWindowsDefDnsServers :: CString -> Int -> IO Word32
 getDefaultDnsServers _ = do
-  res <- peek =<< getWindowsDefDnsServers
-  case dnsError res of
-    0 -> return $ map T.unpack (T.splitOn "," (T.pack (dnsAddresses res)))
-    _ -> do
-      -- TODO: Do proper error handling here.
-      return mempty
+  allocaBytes 128 $ \cString -> do
+     res <- getWindowsDefDnsServers cString 128
+     case res of
+       0 -> do
+         addresses <- BS.pack <$> peekCString cString
+         return $ map BS.unpack (filter (not . BS.null) . BS.split ',' $ addresses)
+       _ -> do
+         -- TODO: Do proper error handling here.
```

Comment:
@rockbmb For the scope of this PR I think I would like to stick with returning `mempty` in case we have an error, as adding more stuff could delay merging the PR. In the future - maybe - all we need is to add a separate type constructor to the `DNSError` type we have already, and throw a proper domain-specific exception. My 2 cents anyway :)

Reply:
@adinapoli-iohk alright, sounds great.